### PR TITLE
Feat: Allow streaming responses from the AI

### PR DESCRIPTION
**Pull request by areejfnaqvi**
## Description

This PR corresponds to the following [Task](https://www.notion.so/conradlabshq/Enable-Streaming-Responses-e7d3fed2331749589531214cd9587d5f?pvs=4)

The AI now returns streaming responses

### CODE REVIEW

File: components/prompt-form.tsx [hunk 1]
```diff
@@ -123,62 +123,86 @@ export function PromptForm({
...
-        message: assistantResponse,
-        role: Roles.assistant
+        dispatch(
+          addMessage({
+            id: newAssistantChatId,
+            message: assistantResponse,
+            role: Roles.assistant
+          })
+        )
+      } else if (message.event === 'thread.run.requires_action') {
```
Comment (AwaisKamran) [inline components/prompt-form.tsx:151]: Same comment here https://github.com/AwaisKamran/custom-canopy-chatbot-app/pull/11/files#r1886613683

File: components/prompt-form.tsx [hunk 2]
```diff
@@ -253,42 +278,48 @@ export function PromptForm({
...
-      const reversedMessages = messages.data.reverse()
 
-      const assistantResponse =
-        // @ts-ignore
-        reversedMessages[reversedMessages.length - 1].content[0].text.value
+    let assistantResponse = ''
+    const newAssistantChatId = nanoid()
+    for await (const message of stream) {
+      if (
+        message.event === 'thread.message.delta' &&
```
Comment (AwaisKamran) [inline components/prompt-form.tsx:294]: Same comment here https://github.com/AwaisKamran/custom-canopy-chatbot-app/pull/11/files#r1886613683

File: components/prompt-form.tsx [hunk 3]
```diff
@@ -253,42 +278,48 @@ export function PromptForm({
...
-        session?.user?.id as string
-      )) as Chat
+        dispatch(
+          addMessage({
+            id: newAssistantChatId,
+            message: assistantResponse,
+            role: Roles.assistant
+          })
+        )
+      } else if (message.event === 'thread.message.completed') {
```
Comment (AwaisKamran) [inline components/prompt-form.tsx:309]: Same comment here https://github.com/AwaisKamran/custom-canopy-chatbot-app/pull/11/files#r1886613683

File: components/prompt-form.tsx (hunk 1 above)
Comment (areejfnaqvi) [reply components/prompt-form.tsx]: done

File: components/prompt-form.tsx (hunk 2 above)
Comment (areejfnaqvi) [reply components/prompt-form.tsx]: done

File: components/prompt-form.tsx (hunk 3 above)
Comment (areejfnaqvi) [reply components/prompt-form.tsx]: done

File: components/prompt-form.tsx [hunk 4]
```diff
@@ -253,42 +278,48 @@ export function PromptForm({
...
+        console.log('Tool outputs submitted successfully')
```
[the remark this answers is not in the document]
Comment (areejfnaqvi) [reply components/prompt-form.tsx]: done

**Review by AwaisKamran — APPROVED** [review]

**Pull request merged**